DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -408,181 +552,62 @@ function Test-RuleProperties
 
     $desiredConfigurationMatch = $true
 
-    if ($Name -and ($FirewallRule.Name -ne $Name))
+    # Loop throug the ParameterList array and compare the source
+    # with the value of each parameter. If different then
+    # set $desiredConfigurationMatch to false.
+    foreach ($p in $ParameterList)
     {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'Name',$FirewallRule.Name,$Name
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($DisplayName -and ($FirewallRule.DisplayName -ne $DisplayName))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'DisplayName',$FirewallRule.DisplayName,$DisplayName
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Group -and ($FirewallRule.Group -ne $Group))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'Group',$FirewallRule.Group,$Group
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Enabled -and ($FirewallRule.Enabled.ToString() -ne $Enabled))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'Enabled',$FirewallRule.Enabled.ToString(),$Enabled
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Action -and ($FirewallRule.Action -ne $Action))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'Action',$FirewallRule.Action,$Action
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Profile)
-    {
-        [String[]] $networkProfileinRule = $FirewallRule.Profile.ToString() -replace(' ', '') -split(',')
-
-        if ($networkProfileinRule.Count -eq $Profile.Count)
+        $ParameterSource = (Invoke-Expression -Command "`$($($p.source))")
+        $ParameterNew = (Invoke-Expression -Command "`$$($p.name)")
+        switch ($p.type)
         {
-            foreach($networkProfile in $Profile)
-            {
-                if (-not ($networkProfileinRule -contains $networkProfile))
+            'String' {
+                # Perform a plain string comparison.
+                if ($ParameterNew -and ($ParameterSource -ne $ParameterNew))
                 {
                     Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                         $($LocalizedData.PropertyNoMatchMessage) `
-                            -f 'Profile',($networkProfileinRule -join ','),($Profile -join ',')
+                            -f $p.Name,$ParameterSource,$ParameterNew
                         ) -join '')
                     $desiredConfigurationMatch = $false
```

Comment:
That's a really good point. I guess it depends if we want to iterate over each test even it fails to write in the log each parameter that mismatches. For debugging of resources this might be useful, but I doubt it is critical. I could go either way with this one. Your call :)